hack/test-cmd.sh

```diff
@@ -238,6 +238,9 @@ fi
 PATH="${KUBE_OUTPUT_HOSTBIN}":$PATH
 
 kube::log::status "Checking kubectl version"
+
+# Note that this fails on a dirty commit.  To avoid that from happening, .gitignore
```

Comment:
oh, I bet I know why.

In CI, we first do [`hack/build-go.sh`](https://github.com/kubernetes/kubernetes/blob/537602d2965806dca6b168a80488b0e75b33b016/hack/jenkins/test-dockerized.sh#L55), then [`go install ./cmd/...`](https://github.com/kubernetes/kubernetes/blob/537602d2965806dca6b168a80488b0e75b33b016/hack/jenkins/test-dockerized.sh#L56). So maybe between those two calls we made the client dirty?

Though the bindata generation happens before anything is built, so I'm still confused.